packages/flutter/test/services/autofill_test.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' show utf8;
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('TextInput message channels', () {
+    FakeTextChannel fakeTextChannel;
+    FakeAutofillScope scope;
+
+    setUp(() {
+      fakeTextChannel = FakeTextChannel((MethodCall call) async {});
+      TextInput.setChannel(fakeTextChannel);
+      scope ??= FakeAutofillScope();
+      scope.autofillClients = <AutofillClient>[];
+    });
+
+    tearDown(() {
+      TextInputConnection.debugResetId();
+      TextInput.setChannel(SystemChannels.textInput);
+    });
+
+    test('text input client handler responds to reattach with setClient', () async {
```

Comment:
Could you add a comment explaining why this is being tested or how this ties into autofill at a high level?

Reply:
Sorry I forgot to change the test description.